pathfinder/tests/test_mocked_integration.py

```diff
@@ -0,0 +1,335 @@
+
+    # Now initialize some channels in this network.
+    # The tuples in channel_descriptions define the following:
+    # (
+    #     p1_index,
+    #     p1_deposit,
+    #     p1_transferred_amount,
+    #     p1_fee,
+    #     p2_index,
+    #     p2_deposit,
+    #     p2_transferred_amount,
+    #     p2_fee
+    # )
+    # Topology:
+    #       /-------------\
+    # 0 -- 1 -- 2 -- 3 -- 4    5 -- 6
+    #  \-------/
+
+    channel_descriptions = [
+        (0, 100,  20, 0.0010, 1,  50,  10, 0.0015),  # capacities  90 --  60
+        (1,  40,  10, 0.0008, 2, 130, 100, 0.0012),  # capacities 130 --  40
+        (2,  90,  10, 0.0007, 3,   0,   0, 0.0010),  # capacities  80 --  10
+        (3,  50,  20, 0.0011, 4,  50,  20, 0.0011),  # capacities  50 --  50
+        (0,  40,  40, 0.0015, 2,  80,   0, 0.0025),  # capacities   0 -- 120
+        (1,  30,  10, 0.0100, 4,  40,  15, 0.0018),  # capacities  35 --  35
+        (5, 500, 900, 0.0030, 6, 750, 950, 0.0040),  # capacities 550 -- 700
+    ]
+    for channel_id, (
+        p1_index,
+        p1_deposit,
+        p1_transferred_amount,
+        p1_fee,
+        p2_index,
+        p2_deposit,
+        p2_transferred_amount,
+        p2_fee
+    ) in enumerate(channel_descriptions):
+        network_listener.emit_event(dict(
+            address=token_network_address,
+            name='ChannelOpened',
+            args=dict(
+                channel_identifier=channel_id,
+                participant1=addresses[p1_index],
+                participant2=addresses[p2_index]
+            )
+        ))
+
+        network_listener.emit_event(dict(
+            address=token_network_address,
+            name='ChannelNewDeposit',
+            args=dict(
+                channel_identifier=channel_id,
+                participant=addresses[p1_index],
+                total_deposit=p1_deposit
+            )
+        ))
+
+        network_listener.emit_event(dict(
+            address=token_network_address,
+            name='ChannelNewDeposit',
+            args=dict(
+                channel_identifier=channel_id,
+                participant=addresses[p2_index],
+                total_deposit=p2_deposit
+            )
+        ))
+
+    # now there should be seven channels
+    assert len(token_network.channel_id_to_addresses.keys()) == 7
+
+    # Now close all channels
+    for channel_id, (
+        p1_index,
+        p1_deposit,
+        p1_transferred_amount,
+        p1_fee,
+        p2_index,
+        p2_deposit,
+        p2_transferred_amount,
+        p2_fee
+    ) in enumerate(channel_descriptions):
+        network_listener.emit_event(dict(
+            address=token_network_address,
+            name='ChannelClosed',
+            args=dict(
+                channel_identifier=channel_id,
+                closing_participant=addresses[p1_index]
+            )
+        ))
+
+    # there should be no channels
+    assert len(token_network.channel_id_to_addresses.keys()) == 0
+
```

Comment:
For an integration test, I'd suggest to also test if the `get_paths` method works, as it is the core functionality of the PFS.

Reply:
Done